### ISSUE TITLE
issue-closer Github Action closes valid issues

### ISSUE THREAD

**Issue by j1elo** [roots/issue-closer-action#1]
* [x] I've read the [guidelines for Contributing to Roots Projects](https://github.com/roots/.github/blob/master/CONTRIBUTING.md)
* [x] This request isn't a duplicate of an existing issue
* [x] I've read the [docs](https://roots.io/documentation/) and followed them (if applicable)
* [x] This is not a personal support request that should be posted on the [Roots Discourse](https://discourse.roots.io/) community

## Description

The `roots/issue-closer@v1.1` Github Action fails regexp match against a correct issue body.

## Steps to reproduce

1. Created a new workflow using `roots/issue-closer@v1.1`
2. Open issue with this body: https://github.com/Kurento/bugtracker/issues/480
3. Action runs with these debug logs:

```
Run roots/issue-closer@v1.1
##[debug]Matching against pattern /^.*description.*reproduce.*$/
##[debug]Body did not match
```

I've also tested with a pattern just like the example one shown in the README, without the begin and end anchors, but it also fails matching:

```
Run roots/issue-closer@v1.1
##[debug]Matching against pattern /.*description.*reproduce.*/
##[debug]Body did not match
```

**Expected behavior:**

Given that the issue body contains both words `description` and `reproduce`, I expected the issue-closer action to not close the issue.

**Actual behavior:** [What actually happens]

issue-closer is not able to match the provided regex, and closes an otherwise valid issue.

**Reproduces how often:** [What percentage of the time does it reproduce?]

100%

## Versions

`roots/issue-closer@v1.1`

## Additional information

The required key words to consider an issue as valid are in different lines. Given that the `issue.body` payload comes from Github in form of a single string, I expected that this wouldn't matter. However, it's possible that some pre-processing and cleaning of the body string must be done before Regexp matching (I am no JS programmer so I don't know the ins and outs of the source code)

**Comment by j1elo** [comment]
The problem is that `RegExp` is created without flags, but the flag `s` would be needed for multi-line suport:

https://github.com/roots/issue-closer-action/blob/95665902c8ad8808e378371ef57fbedbf99ba744/src/main.ts#L58-L62

so it is impossible to define a pattern such as `/line1.*line2/gs`, which is needed to cover multiple lines of the body.

**Comment by swalkinshaw** [comment]
Sorry missed this. Wasn't subscribed to my own repo 🙈 

@j1elo if its as simple as that, do you want to make that change?

**Comment by joey-ma** [comment]
I'm no RegExp expert, but am curious if this would work / serve your purpose. 
Would be happy to hear back from you @j1elo 

For you: 
```yml
    steps:
    - name: Autoclose issues that did not follow issue template
      uses: roots/issue-closer@v1.1
      with:
        repo-token: ${{ secrets.GITHUB_TOKEN }}
        issue-close-message: "@${issue.user.login} this issue was automatically closed because it did not follow the issue template"
        issue-pattern: ".*description[^]*reproduce.*" # here, if it doesn't work, try "yes[^]*day"?
```

From my own experiments:
```js
const s = 'Please yes\nmake my day!'; // a 2-line string

// apparently .* -> any char except new line
console.log(s.match(/.*yes.*day.*/)); // Returns null
console.log(s.match(/yes.*day/)); // Returns null

console.log(s.match(/.*yes|day.*/)); // '|' -> either or
// Returns ["Please yes"]

console.log(s.match(/yes[^]*day/)); //'[^]*' -> using RegExp on multiple lines
// Returns ["yes make my day"]

console.log(s.match(/.*yes[^]*day.*/)); //'[^]*' + '.*'-> still works, kind of?
// Returns ["Please yes make my day!"]
```

Resource: 
- [Using regular expression on multiple lines @ MDN](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/RegExp#using_regular_expression_on_multiple_lines)
- [Regulex](https://jex.im/regulex/#!flags=&re=.*description%5B%5E%5D*reproduce.*)
